recipes-ni/initscripts-nilrt/files/nisetupkernelconfig

```diff
@@ -16,7 +16,16 @@ do
   echo 1 > $x
 done 2> /dev/null
 
-# Set affinity for running IRQ threads to Core-0
-grep -l irq/ /proc/*/comm | cut -d/ -f3 | xargs -n1 taskset -p 1 >/dev/null
+# Set affinity for running IRQ threads to Core-0 immediately.
+# Setting /proc/irq/default_smp_affinity and /proc/irq/*/smp_affinity as done above will
+# eventually cause the kernel to change the affinity for these threads automatically, but
+# not until the next time those threads are scheduled. We want to remove the possibility
+# of that happening during jitter-sensitive code in a user application.
+PIDS=`grep -l irq/ /proc/*/comm 2>/dev/null | cut -d/ -f3`
+if [ -z "$PIDS" ] ; then
```

Comment:
Should this really be an error? I don't understand why. This seems to be the only difference between existing code and new code, other than the comment itself?

Reply:
Yeah; I agree that this doesn't imply an error.

Reply:
Should I change the msg to "No IRQ threads found" and exit 0? [Ref](https://review-board.natinst.com/r/138441/)

Reply:
@billpittman What do you think?

Reply:
Ok I just tried the original and don't see an error when there are no irq threads - so not sure why we added it.

Reply:
Ok, I think we can keep the original file with the updated comment unless @gratian objects.

Reply:
I'll defer to @billpittman since he's the original author. My guess is the error is there to catch a case where this script is not working e.g. the kernel changed the names of irq threads or we misconfigured the kernel and interrupts are not threaded.

Reply:
Definitely does not need to be an error and I think Gratian instincts about the error being there in the first place are correct.  You could probably make a case for removing it all together seeing as it will be corrected eventually and with this running at startup (and the suggested practice of warm-up iterations), there are multiple opportunities for the threads/IRQs to be scheduled/re-scheduled.

Reply:
Changed error to warning and exit 0